src/newViews/Tagging/Sources/SourcesTable/index.tsx

```diff
@@ -0,0 +1,279 @@
+    TableHeaderCellProps,
+    createStringColumn,
+    Tag,
+    TagProps,
+    DateOutput,
+    DateOutputProps,
+    Link,
+    LinkProps,
+    Checkbox,
+    CheckboxProps,
+} from '@the-deep/deep-ui';
+import { IoCheckmarkCircleOutline } from 'react-icons/io5';
+import { VscLoading } from 'react-icons/vsc';
+import { MultiResponse, Lead } from '#typings';
+import { useRequest } from '#utils/request';
+import _ts from '#ts';
+
+import Actions, { Props as ActionsProps } from './Actions';
+import { FormType as Filters } from '../SourcesFilter';
+
+import styles from './styles.scss';
+
+const leadsKeySelector: (d: Lead) => number = d => d.id;
+
+const statusIconMap: Record<Lead['status'], ReactNode> = {
+    pending: <VscLoading />,
+    validated: <IoCheckmarkCircleOutline />,
+    processed: null,
+};
+const statusVariantMap: Record<Lead['status'], 'complement2' | 'accent' | 'complement1'> = {
+    pending: 'complement2',
+    validated: 'accent',
+    processed: 'complement1',
+};
+const statusLabelMap: Record<Lead['status'], string> = {
+    pending: 'In Progress',
+    validated: 'Validated',
+    processed: 'Tagged',
+};
+
+const maxItemsPerPage = 10;
+
+interface Props {
+    className?: string;
+    projectId: number;
+    filters?: Filters;
+}
+
+function SourcesTable(props: Props) {
+    const {
+        className,
+        projectId,
+        filters,
+    } = props;
+
+    const [activePage, setActivePage] = useState<number>(1);
+    const [selectedIds, setSelectedIds] = useState<number[]>([]);
+
+    const leadsRequestQuery = useMemo(() => ({
+        offset: (activePage - 1) * maxItemsPerPage,
+        limit: maxItemsPerPage,
+    }), [activePage]);
+
+    const leadsRequestBody = useMemo(() => ({
+        ...filters,
+        project: projectId,
+    }), [projectId, filters]);
+
+    const {
+        pending: leadsGetPending,
+        response: leadsResponse,
+        retrigger: getLeads,
+    } = useRequest<MultiResponse<Lead>>({
+        url: 'server://v2/leads/filter/',
+        query: leadsRequestQuery,
+        method: 'POST',
+        body: leadsRequestBody,
+        failureHeader: _ts('projectEdit', 'frameworkDetails'),
+    });
+
+    const handlePageChange = useCallback((page: number) => {
+        setActivePage(page);
+        getLeads();
+        setSelectedIds([]);
+    }, [getLeads]);
+
+    const handleSelectAll = useCallback((value: boolean) => {
+        if (value) {
+            const ids = leadsResponse?.results.map(v => v.id) ?? [];
+            setSelectedIds(ids);
+        } else {
+            setSelectedIds([]);
+        }
+    }, [leadsResponse]);
+
+    const handleSelection = useCallback((value: boolean, name: number) => {
+        if (value) {
+            setSelectedIds(ids => ([...ids, name]));
+        } else {
+            setSelectedIds(ids => ids.filter(v => v !== name));
+        }
+    }, []);
+
+    const handleEdit = useCallback(() => {
+        console.warn('handleEdit', handleEdit);
```

Comment:
Use edit modal that we have right now.
Maybe re-create the modal as well?
If too time consuming, lets create an issue for that.

Reply:
@AdityaKhatri let's create a new issue for all the actions on the sources table.